Add Phase I heavy-ion relval wf (take 2)

mandrenguyen opened the pull request:
Same as #19053, which I hosed.  Sorry @perrotta there were a ton of conflicts, so I decided to start over. 

Verbatim:
Adding MC workflows to test --era Run2_2017 with --scenario Heavyions.
Adding wfs are 150, 150.1, 150.2 and 150.3
The latter 3 are migrated from Run 1 wfs 140.1, 140.2 and 140.4
All Hydjet MinBias wfs have been changed to Hydjet impact parameter = 12 fm, which will take care of the memory limitations in running heavy-ion events in relval.
The event overlay wfs (300-302) have not been migrated yet. This will wait for when b=12 GEN-SIM has been produced in a pre-release relval.

The validation and DQM sequences also had to be brought into sync with their pp Phase I equivalents.
To get these sequences to run I modified the heavy-ion HCAL local reco such that it now runs both hbheprereco and hbhereco, as is the case in pp.
I'm not really sure why this was ever different in the first place though.
Feedback on this would be appreciated.

@igv4321 @hatakeyamak @deguio please take note of comments from @kpedro88

mandrenguyen committed:
Add Phase I HI relval wf

Conflicts:
	Configuration/PyReleaseValidation/python/relval_steps.py

mandrenguyen committed:
Add Phase I HI relval wf

Conflicts:
	Configuration/PyReleaseValidation/python/relval_steps.py

mandrenguyen committed:
Add missing gen fragment

cmsbuild commented:
A new Pull Request was created by  @mandrenguyen  for master.

It involves the following packages:

Configuration/Generator
Configuration/PyReleaseValidation
Configuration/StandardSequences
DQM/SiPixelCommon
RecoHI/Configuration
RecoHI/HiJetAlgos
RecoHI/HiTracking


@perrotta, @dmitrijus, @kkousour, @perrozzi, @kpedro88, @fabozzi, @cmsbuild, @franzoni, @thuer, @slava77, @GurpreetSinghChahal, @vanbesien, @govoni, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @jazzitup, @hdelanno, @makortel, @felicepantaleo, @GiacomoSguazzoni, @miheejo, @rovere, @VinInn, @Martin-Grunewald, @yslai, @echapon, @yenjie, @fioriNTU, @kurtejung, @idebruyn, @ebrondol, @threus, @dgulhan, @yetkinyilmaz this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2fef0221db464589346d449d59918b52ac689c38
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20321/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20321/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3414 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1837605
 - DQMHistoTests: Total failures: 19295
 - DQMHistoTests: Total nulls: 65
 - DQMHistoTests: Total successes: 1818072
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

dmitrijus commented:
+1

perrotta commented:
This PR still needs to reach a conclusion about the issue on HCAL local reco raised in https://github.com/cms-sw/cmssw/pull/19098#issuecomment-306280219

hatakeyamak commented:
@igv4321 this is a little beyond me for review. Would you mind taking a look?

igv4321 commented:
Frankly, I do not know either. I am not deeply familiar with HI reconstruction and can't really say whether they need HBHEIsolatedNoiseReflagger or not. One way to find out would be to try it both ways and see what happens...

mandrenguyen commented:
@igv4321 
Is there any good reason to believe this Isolated Noise Reflagger would break down at high occupancy in a non-trivial way (meaning beyond just failing to find noise efficiently)?  
Aside from that the large occupancy I don't think there's anything fundamentally special about the HI reconstruction as far as the HCAL is concerned. 
It would be nice to understand whether someone consciously decide that the HI reconstruction should include this or whether this was accidentally inherited from some deprecated sequence.
From the HI side this was not intentional AFAIK.

igv4321 commented:
I forwarded this discussion to the original author of the HBHEIsolatedNoiseReflagger module

johnpaulchou commented:
My _expectation_ is that the isolated noise filter will simply become inefficient in HI collisions.  One could certainly re-tune the cuts to make it more efficient, but it should not be harmful insofar as it will not identify good collisions as noise.

That said, that is only my expectation, and I haven't tested it on HI data.  It was tuned on pp collisions in relatively high in-time PU scenarios.  Throwing some HI MC at the filter will give you a sense of the over-efficiency, and you may find it is still efficient for noise, especially for peripheral collisions.

mandrenguyen commented:
Thanks @johnpaulchou  
We can take a look in MC.  Dumb question, but what exactly would you look at to convince yourself this is safe to run?

mandrenguyen commented:
@johnpaulchou  Sorry for the dumb question, but can you clarify exactly which flag I should be looking at?  HBHEIsolatedNoiseReflagger is not itself an EDFilter, but looking in FWLite I do see this field:
HBHERecHitsSorted_hbhereco__RECO.obj.obj.flags()
This looks to be filled on a per rechit level, not at an event level.
In 140 events 10123 (out of 188172) are flagged with a non-zero value, compared to 10119 from the comparable "hbhepreco" collection, which corresponds to the EDProducer HBHEPhase1Reconstructor
Does that mean the isolated noise producer flagged 10123 - 10119 = 4 rechits?
Is there an event level flag that I'm missing or is there a subsequent module which produces this?

slava77 commented:
I see no changes in jenkins comparisons for the run-1 workflow (140.53).
So, whatever flagging/reflagging is happening, it is not used in monitored quantities in HI reconstruction.

Also, the preceding discussion mentions filters.
I do not recall MET filters be a part of HI reco workflows or related selections.
@mandrenguyen 
please clarify.

If that's correct and applies just as well to the phase-1 workflow proposed here, I think that the reco part of this PR is fine for HI
and the discussion on HCAL reflagging and filters can move to the future feature developments.

mandrenguyen commented:
@slava77 I suspect you've gone over my head.  I confirm that in the latest 9_2_X IB, after merge-topic of this PR, I see that HBHEIsolatedNoiseReflagger is included in the reconstruction sequence.  MET filters are not part of the heavy-ion reconstruction sequence.  I take your comment to imply that it's MET filters who are the consumers of this noise reflagger.   If I'm reading your comment correctly, and this is basically doing nothing for us, then I would still keep it just to stay synced with the pp reco, and in case we decide to use it in the future.  This assumes this module is not resource intensive in any way.

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/19089	2fef022
- the reco part is rather transparent: switch to the same hbhereco input in HI collisions as the one used in pp. This allows to streamline the configuration and also benefit from the new information now available
- jenkins tests pass and comparisons with baseline show no differences
- I'm estimating that the CPU cost will be up by about 1% on somewhat light "hard probes" or similar events (like in 2011 wf 140.53 in events with ~3K reconstructed tracks); the time increase is dominated by the trackExtrapolator module

davidlange6 commented on Configuration/PyReleaseValidation/python/relval_steps.py:
```diff
@@ -628,25 +628,30 @@ def genS(fragment,howMuch):
+#hiDefaults2018=merge([hiAlca2018,{'--scenario':'HeavyIons','-n':2,'--beamspot':'Realistic50ns13TeVCollision','--geometry':'DB:extended'}])
+hiDefaults2011=merge([hiAlca2011,{'--scenario':'HeavyIons','-n':2}])
+hiDefaults2015=merge([hiAlca2015,{'--scenario':'HeavyIons','-n':2}])
+hiDefaults2018=merge([hiAlca2018,{'--scenario':'HeavyIons','-n':2,'--geometry':'DB:Extended'}])
```
hi @mandrenguyen - DB:Extended is the default - remove it here.

davidlange6 commented on Configuration/PyReleaseValidation/python/relval_steps.py:
```diff
@@ -628,25 +628,30 @@ def genS(fragment,howMuch):
-hiAlca = {'--conditions':'auto:run2_mc_hi', '--era':'Run2_2016,Run2_HI'}
 hiAlca2011 = {'--conditions':'auto:run1_mc_hi'}
+hiAlca2015 = {'--conditions':'auto:run2_mc_hi', '--era':'Run2_2016,Run2_HI'}
+hiAlca2018 = {'--conditions':'auto:phase1_2017_realistic', '--era':'Run2_2017'}
```
there is a 2018 geometry/workflow - should we not use it here?

mandrenguyen replied:
@davidlange6 Indeed this was brought up, and maybe we should.  Since we're currently reviving the HI reco, my thinking was better to use the stable 2017 geometry for the moment, and switch to 2018 detector at a later point.  I don't have a good feeling for how stable the 2018 geometry is.    Would your advice be to switch to the 2018 detector now?

slava77 replied:
The proper "2018" is not currently necessarily best supported.
 @kpedro88 may know better about the state of phase-1 HCAL.

kpedro88 replied:
The 2018 workflow still works fine, but the GT hasn't necessarily been kept up to date with all the latest tags in the 2017 queue.

kpedro88 commented:
@mandrenguyen needs rebase
http://cms-sw.github.io/tutorial-resolve-conflicts.html

cmsbuild commented:
Pull request #19089 was updated. @perrotta, @vazzolini, @dmitrijus, @kmaeshima, @perrozzi, @kpedro88, @fabozzi, @cmsbuild, @kkousour, @franzoni, @thuer, @slava77, @GurpreetSinghChahal, @vanbesien, @govoni, @davidlange6 can you please check and sign again.

mandrenguyen committed:
Update relval_steps.py

Remove extended geometry flag, which is now default, and remove commented out lines

cmsbuild commented:
Pull request #19089 was updated. @perrotta, @vazzolini, @dmitrijus, @kmaeshima, @perrozzi, @kpedro88, @fabozzi, @cmsbuild, @kkousour, @franzoni, @thuer, @slava77, @GurpreetSinghChahal, @vanbesien, @govoni, @davidlange6 can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: d744eee0880cb336066b8462d814e6b18d4b4292
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20725/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/f67a2be8c9a695a868545945e6f113a0053c576d
https://github.com/cms-sw/cmssw/commit/18a30250de2d9a78c330e7f298468c0fcb925784
https://github.com/cms-sw/cmssw/commit/8e25f786011df96c424fa10cc35ba055774d1160
https://github.com/cms-sw/cmssw/commit/f9fc63bd65844762ab69c877c9c2ca2e63af180f
https://github.com/cms-sw/cmssw/commit/a10a86eac334b6589b645996351bba3129f8524d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20725/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20725/git-merge-result

cmsbuild commented:
Comparison job queued.

mandrenguyen committed:
Specifying that a problematic string is an InputTag

mandrenguyen committed:
Specifying that a problematic string is an InputTag

cmsbuild commented:
Pull request #19089 was updated. @perrotta, @vazzolini, @dmitrijus, @kmaeshima, @perrozzi, @kpedro88, @fabozzi, @cmsbuild, @kkousour, @franzoni, @thuer, @slava77, @GurpreetSinghChahal, @vanbesien, @govoni, @davidlange6 can you please check and sign again.

mandrenguyen commented:
@davidlange6 (or anyone with the rights):  Despite passing Jenkins, the wf 150 reco driver command was crashing due to some unrelated change in tracking that came from an unrelated commit that was merged since this PR was made.  I had to add a one-liner to fix this.  Please gimme another 'please test'

davidlange6 commented:
please test

> On Jun 19, 2017, at 4:49 PM, mandrenguyen <notifications@github.com> wrote:
> 
> @davidlange6 (or anyone with the rights): Despite passing Jenkins, the wf 150 reco driver command was crashing due to some unrelated change in tracking that came from an unrelated commit that was merged since this PR was made. I had to add a one-liner to fix this. Please gimme another 'please test'
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f28e8cb5c7658e63a69b85615635b7356e4b88da
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20754/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20725/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1803136
 - DQMHistoTests: Total failures: 111
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1802859
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

slava77 commented on RecoHI/HiMuonAlgos/python/HiRegitMuonPixelPairStep_cff.py:
```diff
@@ -52,7 +52,7 @@
 )
 hiRegitMuPixelPairStepSeeds     = RecoTracker.IterativeTracking.PixelPairStep_cff.pixelPairStepSeeds.clone(
-    seedingHitSets = "hiRegitMuPixelPairStepHitDoublets"
+    seedingHitSets = cms.InputTag("hiRegitMuPixelPairStepHitDoublets")
```
was this type specialization required?
It's better not to specify the type if the parameter is meant to be defined already

mandrenguyen replied:
@slava77 Without this type specialization the driver no longer runs, since the changes made here:  https://github.com/cms-sw/cmssw/commit/f9cefe2826f8be81c8b15cd79e57319f4c286d71#diff-528a13022ace4a870beca2a390be12c3R143

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/19089	f28e8cb
- changes since the last reco signoff are minimally affecting reco: 
    - rebase to catch up with HI PV refit
    - somewhat unnecessary but OK update to specify the type of seedingHitSets in  RecoHI/HiMuonAlgos/python/HiRegitMuonPixelPairStep_cff.py 
- jenkins tests pass and comparisons show no differences

kpedro88 commented:
+1

slava77 commented:
On 6/19/17 9:40 PM, mandrenguyen wrote:
> *@mandrenguyen* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In RecoHI/HiMuonAlgos/python/HiRegitMuonPixelPairStep_cff.py
> <https://github.com/cms-sw/cmssw/pull/19089#discussion_r122879648>:
> 
>> @@ -52,7 +52,7 @@
>      clusterCheck = "hiRegitMuClusterCheck",
>  )
>  hiRegitMuPixelPairStepSeeds     = RecoTracker.IterativeTracking.PixelPairStep_cff.pixelPairStepSeeds.clone(
> -    seedingHitSets = "hiRegitMuPixelPairStepHitDoublets"
> +    seedingHitSets = cms.InputTag("hiRegitMuPixelPairStepHitDoublets")
> 
> @slava77 <https://github.com/slava77> Without this type specialization
> the driver no longer runs, since the changes made here:
> f9cefe2#diff-528a13022ace4a870beca2a390be12c3R143
> <https://github.com/cms-sw/cmssw/commit/f9cefe2826f8be81c8b15cd79e57319f4c286d71#diff-528a13022ace4a870beca2a390be12c3R143>

I see, this is for phase-1.
But then the change in this PR is meaningless.
because in phase-1 the pixelPairStepSeeds doesn't have a seedingHitSets
parameter.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19089#discussion_r122879648>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbqO3rVCuKkcaBV1CtCvoSNQvAS_aks5sF00jgaJpZM4NvafL>.
>

slava77 commented:
I dropped the reco +1 because it looks like the changes made are incomplete/inconsistent

mandrenguyen commented:
@slava77 You lost me.  There was a +1 before the fix, but the driver wasn't running.  Now it runs, but this is not approved?  What is your preferred solution?

slava77 commented:
On 6/19/17 10:14 PM, mandrenguyen wrote:
> @slava77 <https://github.com/slava77> You lost me. There was a +1 before
> the fix, but the driver wasn't running. Now it runs, but this is not
> approved? What is your preferred solution?

I found out that my review was incomplete based on the most recent
information.
So, I dropped the "+1".

My preliminary comment is that the "fix" in pixelPairStepSeeds is not
functional,
but it needs to be checked.
you may want to dump the config in your setup and confirm if it makes
desired changes to parameters available to this specific module.

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19089#issuecomment-309646845>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbmfqnlPt6wHa-qGBsaUxXqYXtTgrks5sF1UkgaJpZM4NvafL>.
>

makortel commented:
@mandrenguyen When `trackingPhase1` modifier is active, `pixelPairStepSeeds` is no longer a `SeedCreatorFromRegionConsecutiveHitsEDProducer`, but a `SeedCombiner`. To always get a `SeedCreatorFromRegionConsecutiveHitsEDProducer` you can clone from `pixelPairStepSeedsA`.

mandrenguyen committed:
load parameter from pixelPairStepSeedsA instead of pixelPairStepSeeds

cmsbuild commented:
Pull request #19089 was updated. @kpedro88, @fabozzi, @Martin-Grunewald, @rekovic, @dmitrijus, @govoni, @perrotta, @vanbesien, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @kkousour, @slava77, @mulhearn, @vazzolini, @kmaeshima, @perrozzi, @thuer, @monttj, @franzoni can you please check and sign again.

Martin-Grunewald commented:
-1

many unrelated PRs pulled in with their commits

cmsbuild commented:
Pull request #19089 was updated. @perrotta, @vazzolini, @dmitrijus, @kmaeshima, @perrozzi, @kpedro88, @fabozzi, @cmsbuild, @kkousour, @franzoni, @thuer, @slava77, @GurpreetSinghChahal, @vanbesien, @govoni, @davidlange6 can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: bd06b9907957022b3080bf809536493a9129a250
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20785/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19089/20785/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1803260
 - DQMHistoTests: Total failures: 29439
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1773655
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

slava77 commented on Configuration/PyReleaseValidation/python/relval_steps.py:
```diff
@@ -628,25 +628,28 @@ def genS(fragment,howMuch):
+
+steps['HydjetQ_B12_5020GeV_2011']=merge([{'-n':1,'--beamspot':'RealisticHI2011Collision'},hiDefaults2011,genS('Hydjet_Quenched_B12_5020GeV_cfi',U2000by1)])
+steps['HydjetQ_B12_5020GeV_2015']=merge([{'-n':1,'--beamspot':'RealisticHICollision2015'},hiDefaults2015,genS('Hydjet_Quenched_B12_5020GeV_cfi',U2000by1)])
+steps['HydjetQ_B12_5020GeV_2018']=merge([{'-n':1,'--beamspot':'Realistic50ns13TeVCollision'},hiDefaults2018,genS('Hydjet_Quenched_B12_5020GeV_cfi',U2000by1)])
```
Note that the 2017 workfows have migrated to Realistic25ns13TeVEarly2017Collision by now.
this has to be in sync with the GT (phase1_2017_realistic).
It may be easier to maintain synchronous updates if you derive settings from       what is used in ```upgradeStepDict['GenSimFull'][k]= {'-s' : 'GEN,SIM', ``` at the end of the file.

mandrenguyen replied:
Thanks for catching that.  I propose to take care of this in a subsequent pull request, as soon as this one gets merged.

slava77 commented:
I ran 100 events for the added workflows: 150.2 failed for a not very obvious reason.
The jobs were 6-thread each.

in step3 on the last event in 5 steams out of 6.
```
An exception of category 'FatalRootError' occurred while
   [0] Processing  stream end Run run: 1 stream: 5
   [1] Calling method for module SiPixelClusterSource/'SiPixelClusterSource'
   Additional Info:
      [a] Fatal Root Error: @SUB=Merge
Cannot merge histograms - limits are inconsistent:
 first: pix_bar Occ_roc_online_siPixelDigis_layer_4 (72, -5.000000, 5.200000), second: pix_bar Occ_roc_online_siPixelDigis_layer_4 (72, -5.200000, 5.200000)
```

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/19089	bd06b99
- reco parts look OK; the beam spot settings in SIM look wrong and will need an update, but that's non-reco package settings.
- jenkins tests pass and comparisons with baseline show no differences 
    - the phase-0 reco in HI is not affected
- local check with 150.{0,1,2} show that the workflows run and the set of modules in step3 looks sensible
    - I'm ignoring the issue in SiPixelClusterSource: the module is configured in the same way as in 10224 and should technically work just as well

kpedro88 commented:
+1

slava77 commented:
this PR is still missing signatures from dqm, generators, pdmv, and operations
@dmitrijus, @kkousour, @perrozzi, @fabozzi, @franzoni, @thuer, @GurpreetSinghChahal, @vanbesien, @govoni
please take a look

fabozzi commented:
+1

davidlange6 commented:
merge

Dr15Jones commented:
Workflows 150.1, 150.2 and 150.3 are all failing with an failed `assert`.

https://cms-sw.github.io/relvalLogDetail.html#slc6_amd64_gcc530;CMSSW_9_3_X_2017-06-28-1100

mandrenguyen commented:
I'm not able to reproduce this failure.  When I run 150.3 out of the box in CMSSW_9_3_X_2017-06-28-2300 I get:
150.3_ZEEMM_13_HI+ZEEMM_13_HI+DIGIHI2018+RECOHI2018+HARVESTHI2018 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Thu Jun 29 08:09:54 2017-date Thu Jun 29 08:04:02 2017; exit: 0 0 0 0
1 1 1 1 tests passed, 0 0 0 0 failed

If I look at the log of step 2, I do see this message:
%MSG-w L1GtTrigReport:  L1GtTrigReport:hltL1GtTrigReport  29-Jun-2017 08:06:47 CEST Run: 1 Event: 1
  L1GlobalTriggerReadoutRecord with input tag hltGtDigis has gtFdlVector of size 0
  Invalid L1GlobalTriggerReadoutRecord!
  Event classified as Error

but I also see that for 150, which isn't listed as failed.

@Dr15Jones

davidlange6 commented:
add -i all and you'll reproduce...however, it doesn't seem like this workflow should be configured to use gen-sim from 76x...

> On Jun 29, 2017, at 8:16 AM, mandrenguyen <notifications@github.com> wrote:
> 
> I'm not able to reproduce this failure. When I run 150.3 out of the box in CMSSW_9_3_X_2017-06-28-2300 I get:
> 150.3_ZEEMM_13_HI+ZEEMM_13_HI+DIGIHI2018+RECOHI2018+HARVESTHI2018 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED - time date Thu Jun 29 08:09:54 2017-date Thu Jun 29 08:04:02 2017; exit: 0 0 0 0
> 1 1 1 1 tests passed, 0 0 0 0 failed
> 
> If I look at the log of step 2, I do see this message:
> %MSG-w L1GtTrigReport: L1GtTrigReport:hltL1GtTrigReport 29-Jun-2017 08:06:47 CEST Run: 1 Event: 1
> L1GlobalTriggerReadoutRecord with input tag hltGtDigis has gtFdlVector of size 0
> Invalid L1GlobalTriggerReadoutRecord!
> Event classified as Error
> 
> but I also see that for 150, which isn't listed as failed.
> 
> @Dr15Jones
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

mandrenguyen commented:
@davidlange6 It shouldn't take any input at all, since these are no-PU wfs
I see the input defined here, e.g.,:
relval_steps.py:steps['ZEEMM_13_HIINPUT']={'INPUT':InputInfo(dataSet='/RelValZEEMM_13_HI/%s/GEN-SIM'%(baseDataSetRelease[1],),location='STD')}
and I don't see any further reference to that line.
If I just zap that line, is that sufficient?

davidlange6 commented:
we definitely use the ability to use existing gen-sim as input

likely you just need to add an entry to 
https://github.com/mandrenguyen/cmssw/blob/bd06b9907957022b3080bf809536493a9129a250/Configuration/PyReleaseValidation/python/relval_steps.py#L374
and use that in your workflow

mandrenguyen commented:
@davidlange6 Ah, recycling GEN-SIM, I see.   Ok, then I guess this recycling is failing because I changed these wfs from the 2015 to the 2017 detector.    Will new GEN-SIM be produced with 930pre1?  If so, should I just wait and update the base release to use that as input once it's created?

fabozzi commented:
@mandrenguyen 
Yes we can redo gensim. When 930_pre1 is out, please clarify which HI workflows are needed for every release, and which ones only in 930_pre1 to redo gensim

mandrenguyen commented:
@fabozzi  In 93X we need new GS for 150, 150.1, 150.2 and 150.3.  We need these wfs to be run  for every pre-release, but the GS can be recycled so long as the 2017 simulation is stable.

fabozzi commented:
@mandrenguyen
I have passed this information to L3 validation managers

mandrenguyen commented:
@fabozzi :  930pre1 is out and the requested workflows discussed in this thread were run.  Thanks!  For the rest of the 93X pre-release cycle, it would also be very useful if the heavy-ion embedding wfs (300, 301 and 302) could also be run, using the output of wf 150 as the pile-up dataset.  Feel free to move this off github with the relevant parties in cc.

fabozzi commented:
@mandrenguyen 
OK, I will forward the information to validation L3

fabozzi commented:
@mandrenguyen 
to update the mixing input, I think we have to use:
/RelValHydjetQ_B12_5020GeV_2018/CMSSW_9_3_0_pre1-92X_upgrade2017_realistic_v7-v1/GEN-SIM

in the following lines of relval_steps.py :
https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L973
https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L1578

Also, I think we can get rid of:
https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L677-L678

since those HydjetQ_MinBias have been superceded.

davidlange6 commented:
the proposal is to remove the tests of old HI processing?

> On Jul 11, 2017, at 6:11 AM, fabozzi <notifications@github.com> wrote:
> 
> @mandrenguyen
> to update the mixing input, I think we have to use:
> /RelValHydjetQ_B12_5020GeV_2018/CMSSW_9_3_0_pre1-92X_upgrade2017_realistic_v7-v1/GEN-SIM
> 
> in the following lines of relval_steps.py :
> https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L973
> https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L1578
> 
> Also, I think we can get rid of:
> https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L677-L678
> 
> since those HydjetQ_MinBias have been superceded.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>
